toolchain/private/defs.bzl

```diff
@@ -40,36 +40,52 @@ def zig_tool_path(os):
     else:
         return _ZIG_TOOL_PATH
 
-def target_structs():
+def target_structs(macos_sdk_versions):
     ret = []
     for zigcpu, gocpu in (("x86_64", "amd64"), ("aarch64", "arm64")):
-        ret.append(_target_macos(gocpu, zigcpu))
         ret.append(_target_windows(gocpu, zigcpu))
         ret.append(_target_linux_musl(gocpu, zigcpu))
         for glibc in _GLIBCS:
             ret.append(_target_linux_gnu(gocpu, zigcpu, glibc))
+        for macos_sdk_version in macos_sdk_versions:
+            ret.append(_target_macos(gocpu, zigcpu, macos_sdk_version))
     ret.append(_target_wasm())
     return ret
 
-def _target_macos(gocpu, zigcpu):
-    copts = []
+def _target_macos(gocpu, zigcpu, macos_sdk_version):
+    macos_sdk_opts = [
+        "--sysroot",
+        "external/hermetic_cc_toolchain~~toolchains~macos_sdk_{}".format(macos_sdk_version),
+        "-F",
+        "external/hermetic_cc_toolchain~~toolchains~macos_sdk_{}/Frameworks".format(macos_sdk_version),
+        "-L",
+        "external/hermetic_cc_toolchain~~toolchains~macos_sdk_{}/lib".format(macos_sdk_version),
```

Comment:
I'm not sure why `external/macos_sdk_…` path doesn't work. Is it how the repository was defined? I'm not sure if this is idiomatic bazel to depend on an external path with `~`s, I'd imagine these are 'non-public' paths.

Reply:
These are bzlmod canonical repo paths, which are not guaranteed to be deterministic. You'll want to instead resolve them with some combination of `Label` and `repository_ctx.path`